src/cloud-providers/azure/manager.go

```diff
@@ -18,37 +18,36 @@ func init() {
-	flags.StringVar(&azurecfg.Size, "instance-size", "Standard_DC2as_v5", "Instance size")
-	flags.StringVar(&azurecfg.ImageId, "imageid", "", "Image Id")
-	flags.StringVar(&azurecfg.SubscriptionId, "subscriptionid", "", "Subscription ID")
-	flags.StringVar(&azurecfg.SSHKeyPath, "ssh-key-path", "", "Path to SSH public key")
-	flags.StringVar(&azurecfg.SSHUserName, "ssh-username", "peerpod", "SSH User Name")
-	flags.BoolVar(&azurecfg.DisableCVM, "disable-cvm", false, "Use non-CVMs for peer pods")
-	// Add a List parameter to indicate different types of instance sizes to be used for the Pod VMs
-	flags.Var(&azurecfg.InstanceSizes, "instance-sizes", "Instance sizes to be used for the Pod VMs, comma separated")
-	// Add a key value list parameter to indicate custom tags to be used for the Pod VMs
-	flags.Var(&azurecfg.Tags, "tags", "Custom tags (key=value pairs) to be used for the Pod VMs, comma separated")
-	flags.BoolVar(&azurecfg.EnableSecureBoot, "enable-secure-boot", false, "Enable secure boot for the VMs")
-	flags.BoolVar(&azurecfg.UsePublicIP, "use-public-ip", false, "Assign public IP to the PoD VM and use to connect to kata-agent")
-	flags.IntVar(&azurecfg.RootVolumeSize, "root-volume-size", 0, "Root volume size in GB. Default is 0, which implies the default image disk size")
+	reg := provider.NewFlagRegistrar(flags)
+
+	// Flags with environment variable support
+	reg.StringWithEnv(&azurecfg.ClientId, "clientid", "", "AZURE_CLIENT_ID", "Client Id")
+	reg.StringWithEnv(&azurecfg.ClientSecret, "secret", "", "AZURE_CLIENT_SECRET", "Client Secret")
+	reg.StringWithEnv(&azurecfg.TenantId, "tenantid", "", "AZURE_TENANT_ID", "Tenant Id")
+	reg.StringWithEnv(&azurecfg.SubscriptionId, "subscriptionid", "", "AZURE_SUBSCRIPTION_ID", "Subscription ID")
+	reg.StringWithEnv(&azurecfg.Region, "region", "", "AZURE_REGION", "Region")
+	reg.StringWithEnv(&azurecfg.ResourceGroupName, "resourcegroup", "", "AZURE_RESOURCE_GROUP", "Resource Group")
+	reg.StringWithEnv(&azurecfg.Size, "instance-size", "Standard_DC2as_v5", "AZURE_INSTANCE_SIZE", "Instance size")
+
+	// Flags without environment variable support (pass empty string for envVarName)
+	reg.StringWithEnv(&azurecfg.Zone, "zone", "", "", "Zone")
+	reg.StringWithEnv(&azurecfg.SubnetId, "subnetid", "", "", "Network Subnet Id")
+	reg.StringWithEnv(&azurecfg.SecurityGroupId, "securitygroupid", "", "", "Security Group Id")
+	reg.StringWithEnv(&azurecfg.ImageId, "imageid", "", "", "Image Id")
+	reg.StringWithEnv(&azurecfg.SSHKeyPath, "ssh-key-path", "", "", "Path to SSH public key")
+	reg.StringWithEnv(&azurecfg.SSHUserName, "ssh-username", "peerpod", "", "SSH User Name")
+	reg.BoolWithEnv(&azurecfg.DisableCVM, "disable-cvm", false, "", "Use non-CVMs for peer pods")
+	reg.BoolWithEnv(&azurecfg.EnableSecureBoot, "enable-secure-boot", false, "", "Enable secure boot for the VMs")
+	reg.BoolWithEnv(&azurecfg.UsePublicIP, "use-public-ip", false, "", "Assign public IP to the PoD VM and use to connect to kata-agent")
```

Comment:
This uses `USE_PUBLIC_IP` env, but is in 4 providers, so we could consider making it shared and moving to `main.go`

Reply:
same as before.